### ISSUE TITLE
YapDatabaseCloudKit demo app stops syncing after going offline & online again

### ISSUE THREAD

**Issue by kluku** [yapstudios/YapDatabase#135]
Hi,
I've implemented very simple journal app based on the CloudKitTodo demo and started testing it. It works great when both devices are online, however it doesn't upload changes, if I disable the connection and reenable it. The only solution is to kill the app and restart it. I was pretty sure that it was my mistake... so I installed the original CloudKitTodo app on both devices and reproduced exactly the same scenario: open the app, disable wifi, add new item, reenable wifi - and it never gets synchronized until I kill the app completely and relaunch it. On the bottom it says: Status: resumed, ChangeSets: InFlight(1), Queued(0). Please help me.. the deadline is getting closer : /
![img_0295](https://cloud.githubusercontent.com/assets/118002/5940870/d0a43ba0-a70d-11e4-9112-438ff3060b9e.png)

**Comment by robbiehanson** [comment]
Is Reachability firing properly? You should be able to set a breakpoint in [DatabaseManager reachabilityChanged:] & [CloudKitManager reachabilityChanged:], and these should be getting hit when WiFi comes back up.

**Comment by kluku** [comment]
Yes, both reachability delegates are properly called. To describe more precisely: I start the application in online mode, then go offline (disable wifi connection from the bottom menu, and the cellular connection is disabled all the time). Reachability delegates are fired. Status is still "Resumed". Now I add a new note. Status becomes "Suspended" with suspendCount = 1. I enable wifi using bottom menu, and delegates are fired again. CloudKitManager reachabilityChanged doesn't need to configure cloud kit and it doesn't fetch record changes. DatabaseManager resumes cloud kit extension. On the screen, status becomes "Resumed" again, but in change sets there is InFlight(1) and it stays this way forever... I can disable/enable wifi many times, but it doesnt change anything. The new note is finally uploaded only when I kill the app and relaunch it.

**Comment by robbiehanson** [comment]
Thank you for the detailed 'steps to reproduce' report. I was able to quickly reproduce the problem.

The issue ended up being that, when an upload failed, the YDBCKChangeSet.isInFlight property was not properly reset. So the system acted as if an upload was in progress, and basically just queued all operations from that point forward.

I pushed a fix for you: commit d3ae1c09352bf0efa131570899e04a8d8e880f8d
